sn_auditor/src/dag_db.rs

```diff
@@ -228,39 +229,70 @@ impl SpendDagDb {
                 continue;
             }
 
-            // get a copy of the current DAG
-            let mut dag = { self.dag.clone().read().await.clone() };
-
-            // update it
-            client
-                .spend_dag_continue_from(&mut dag, addrs_to_get, spend_processing.clone(), true)
-                .await;
-
-            // update utxos
-            let new_utxos = dag.get_utxos();
-            utxo_addresses.extend(
-                new_utxos
-                    .into_iter()
-                    .map(|a| (a, Instant::now() + REATTEMPT_INTERVAL)),
-            );
+            #[cfg(not(feature = "dag-collection"))]
+            {
+                if let Some(sender) = spend_processing.clone() {
+                    // crawl DAG
+                    let tasks: Vec<_> = addrs_to_get
+                        .iter()
+                        .map(|a| client.spend_dag_crawl_from(*a, sender.clone()))
+                        .collect();
+                    let res = futures::future::join_all(tasks).await;
+                    let mut new_utxos = BTreeSet::new();
+                    for (r, a) in res.into_iter().zip(addrs_to_get) {
+                        match r {
+                            Ok(utxos) => new_utxos.extend(utxos),
+                            Err(e) => error!("Failed to crawl DAG from {a:?} : {e}"),
+                        }
+                    }
 
-            // write updates to local DAG and save to disk
-            let mut dag_w_handle = self.dag.write().await;
-            *dag_w_handle = dag;
-            std::mem::drop(dag_w_handle);
-            if let Err(e) = self.dump().await {
-                error!("Failed to dump DAG: {e}");
+                    // update utxos
+                    utxo_addresses.extend(
+                        new_utxos
+                            .into_iter()
+                            .map(|a| (a, Instant::now() + REATTEMPT_INTERVAL)),
+                    );
+                } else {
+                    panic!("There is no point in running the auditor if we are not collecting the DAG or collecting data through crawling. Please enable the `dag-collection` feature or provide beta program related arguments.");
```

Comment:
In the future we can add royalties collection as a background task similar to beta reward tracking